lib/config.go

```diff
@@ -111,43 +102,25 @@ func (cfg *Config) check() error {
 // readSettings Reads the .s3deploy.yml file for configuration settings.
 func (cfg *Config) readSettings() error {
 	configFile := cfg.ConfigFile
-	settingsFile := yamlConfig{}
 
 	if configFile == "" {
 		return nil
 	}
 
+	// Configuration file is optional; when it does not exist,
+	// we rely on command flags and global AWS configuration
 	if _, err := os.Stat(configFile); os.IsNotExist(err) {
 		return nil
 	}
 
 	settings, err := ioutil.ReadFile(configFile)
-
-	// No problem if the file doesn't exist; then only rely on command flags
-	if os.IsNotExist(err) {
-		return nil
-	}
-
-	err = yaml.Unmarshal(settings, &settingsFile)
 	if err != nil {
-		return nil
+		return errors.New("could not find the settings file")
 	}
 
-	// Load in settings, but only when the accompanying command flag hasn't been set
```

Comment:
The above sentence should be respected.

Reply:
So if the user choose to add his/her AWS key and secret to the config file, the general AWS credentials?

I tried several times to program that last week, but it's unfortunately outside my current experience to implement. So just checking here if I actually understood correctly what you're saying.

Reply:
Ah I see, thanks for clarifying. I was looking in the wrong direction.

For what you ask, the code works like that for me. This is my test `.s3deploy.yml` file:

~~~
bucket: example.com
key: xcsds
secret: xdsfdsf
region: us-west
source: public
~~~

But I can override these settings with the command flags. When I use the latest commit above (18ac3a0) it goes like so.

With command flags, it uses `us-west` as the region and `example.com` for the bucket name:

~~~
C:\Users\Jos\go\src\github.com\bep\s3deploy>go run main.go -try
s3deploy v2, commit none, built at unknown
This is a trial run, with no remote updates.

Total in 0.35 seconds
error: RequestError: send request failed
caused by: Get https://s3.us-west.amazonaws.com/example.com: dial tcp: lookup s3.us-west.amazonaws.com: no such host
~~~

But with command flags those settings are overruled:

~~~
C:\Users\Jos\go\src\github.com\bep\s3deploy>go run main.go -try -bucket=myexamplewebsite.com -region=europe-512
s3deploy v2, commit none, built at unknown
This is a trial run, with no remote updates.

Total in 0.37 seconds
error: RequestError: send request failed
caused by: Get https://s3.europe-512.amazonaws.com/myexamplewebsite.com: dial tcp: lookup s3.europe-512.amazonaws.com: no such host
~~~

Now it looked for the `myexamplewebsite.com` in the region `europe-512`, *thereby* overriding the settings specified in the `s3deploy.yml` file.